src/NUnitTestExtractor/NUnitTestExtractor.Tests/NUnitTestExtractorTests.cs

```diff
@@ -3,22 +3,309 @@
+        {
+            List<string> dlls = new List<string>();
+            dlls.Add(dll);
+
+            Process process = SetUpProcess(dll, level);
+
+            Assert.That(process.StandardOutput.ReadToEnd(), Is.Empty);
+
+            process.WaitForExit();
+        }
+
+        [TestCase("C:/Git/NTestController/bin/Debug/NTestControllerTests.dll", NUnitTestExtractorApp.Level.Namespace)]
+        public void WritingToStdout_LevelIsNamespace_OnlyNamespaceWritten(string dll, NUnitTestExtractorApp.Level level)
+        {
+            List<string> dlls = new List<string>();
+            dlls.Add(dll);
+
+            Process process = SetUpProcess(dll, level.ToString());
+
+            string line = process.StandardOutput.ReadLine();
+            string result = line.Substring(line.IndexOf("|") + 2);
+
+            Assert.That(!result.Contains("."));
+        }
+
+        [TestCase("C:/Git/NTestController/bin/Debug/NTestControllerTests.dll", NUnitTestExtractorApp.Level.Class)]
+        public void WritingToStdout_LevelIsClass_NamespaceAndClassWritten(string dll, NUnitTestExtractorApp.Level level)
+        {
+            List<string> dlls = new List<string>();
+            dlls.Add(dll);
+
+            Process process = SetUpProcess(dll, level.ToString());
+
+            string line = process.StandardOutput.ReadLine();
+            string result = line.Substring(line.IndexOf("|") + 2);
+
+            //This works as class and namespace are seperated by a period example: namepspace.class
+            Assert.That(result.Contains("."));
+        }
+
+        [TestCase("C:/Git/NTestController/bin/Debug/NTestControllerTests.dll", NUnitTestExtractorApp.Level.Function)]
+        public void WritingToStdout_LevelIsFunction_NamespaceClassAndFunctionGetWritten(string dll, NUnitTestExtractorApp.Level level)
+        {
+            List<string> dlls = new List<string>();
+            dlls.Add(dll);
+
+            Process process = SetUpProcess(dll, level.ToString());
+
+            string line = process.StandardOutput.ReadLine();
+            string result = line.Substring(line.IndexOf("|") + 2);
+
+            //Same deal as test above, function and class separated with a period: namespace.class.function
+            Assert.That(result.Count(p => p == '.'), Is.EqualTo(2));
         }
 
+        [TestCaseSource(typeof(ValidDataTestCases))]
+        public void WritingToStdout_ValidDllAndEveryLevel_ContainsNoDuplicates(string dll, string level)
+        {
+            List<string> dlls = new List<string>();
+            dlls.Add(dll);
+
+            Process process = SetUpProcess(dll, level);
+
+            string line = "";
+
+            List<string> lines = new List<string>();
+
+            while((line = process.StandardOutput.ReadLine())!= null)
+            {
+                lines.Add(line);
+            }
+
+            Assert.That(lines.Count == lines.Distinct().Count());
```

Comment:
Instead of using Assert.That(x == y) it's better to use Assert.AreEqual(x, y).
If it fails, it'll print:
```
Expected: x
But was: y
```
(so the first parameter should be the value you're expecting)